docs/DEVELOPMENT-TIPS-TRICKS.md

```diff
@@ -0,0 +1,184 @@
+
+It includes:
+
+- serverless service manifest file: `serverless.yml` - defines cloud resources
+  and glue between logic of the service
+- AWS lambda function which gets triggered when a new file for the ETL comes
+  in - in this case it's `./src/events/onParseCSV.js`. Without going into too
+  many details here - you can take an example from another service matching the
+  file format (extension) from another service. (CSV for example)
+- Creating boilerplate assets: `test` folder, `webpack.config.js` file, etc. These
+  you possibly already have taking the example service from the previous step.
+
+2.  Grab an example record structure
+
+- Deploy the service with `npx serverless deploy` ran inside the root of the service
+- Trigger the function and `console.log` the incoming `event` parameter in order
+  to generate stubs for event and record. These are later useful for local development.
+  You might want to `JSON.strigify` objects for better readability in CloudWatch logs.
+
+3.  Write a small transform function
+
+- It's usually placed at `src/lib/transform.js`
+- It should include the [flow][5] type for [Project][6]
+- Include [JSDoc][7] comments to all helper functions in the transform in order
+  to expose the smaller transformation steps in a user friendly [API pages][8]
+
+4.  Write a test for the transform function
+
+- It's usually placed at `test/unit/lib/transform.spec.js`
+- Include assertions for matching snapshot
+
+```js
+test('Produces correct JSON output structure', () => {
+  expect(result).toMatchSnapshot();
+});
+```
+
+- Run [jest in watch mode][9] to iterate faster. Each time you edit the file with
+  the transform function, the test runner will restart the test. You might want to
+  use `console.log` to see results of your transform function while you develop it.
+
+### Debugging a lambda function locally
+
+As AWS lambda functions are also known as cloud functions, they are not always easy
+to debug locally. However, making use of the serverless framework, we have some helpers
+to do debugging locally.
+
+1.  In `webpack.config.js`, set a new property if not already set `devtool: 'source-map'`.
+    This will let you to see code for humans, not transpiled one during execution of code
+
+2.  Add a breakpoint in with `debug;` either in lambda function or transform function
+
+3.  Run the following in the CLI
+    `node --inspect-brk ./node_modules/.bin/serverless invoke local -f fooFunction --path eventStub.json`
+
+Where:
+
+- `node --inspect-brk` is the Node.js core inspector protocol which opens a session to Chrome browser
+- `serverless invoke local` is a specific command of the [serverless CLI][11]
+- `fooFunction` is the name of the function as found in the `serverless.yml` file
+- `eventStub` is the JSON file which you can take by using `console.log(JSON.strigify(event))`
+  in the beginning of the lambda function
+
+Open [Node.js debugger your Chrome browser's devtools][12]. You can also use the
```

Comment:
there's something wrong in this sentence

Reply:
Reworked it :)